### PR TITLE
07 add filter pokemon list

### PR DESCRIPTION
**Pull request by bdtkevin**
Style is not definitive..

### CODE REVIEW

**Review by gasparddhi — DISMISSED** [review]
Congratulation very clean work ! I approve

**Comment by gasparddhi** [comment]
Nice PR

**Review by RobinDela — DISMISSED** [review]
Ok

File: src/components/Filters.jsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+          <div className="select-weight">
+            <h2>weight</h2>
+            <select
+              name="weight"
+              id="weightSelect"
+              onChange={(e) => {
+                const selectBox = e.target;
+
+                setWeightFilters(
+                  selectBox.options[selectBox.selectedIndex].value
```
Comment (LoicBrassart) [inline src/components/Filters.jsx]: I may be wrong, but I think `e.target.value` does exactly the same :thinking:

File: src/components/Filters.jsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+          wrapper.classList.toggle('open');
+          e.target.innerHTML =
+            e.target.innerHTML === 'Show Filters'
+              ? 'Hide Filters'
+              : 'Show Filters';
+        }}
+      >
+        Show Filters
+      </button>
```
Comment (LoicBrassart) [inline src/components/Filters.jsx]: This seems... awfully complicated, doesn't it ?

File: src/components/FiltersAbilities.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const handleAbility = (e) => {
+    const selectBox = e.target;
+
+    setAbilityFilters(selectBox.options[selectBox.selectedIndex].value);
```
Comment (LoicBrassart) [inline src/components/FiltersAbilities.jsx]: `setAbilityFilter(e.target.value)`

File: src/components/FiltersTypes.jsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  const [types, setTypes] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    axios
+      .get('https://pokeapi.co/api/v2/type')
+      .then(({ data }) => {
+        setTypes(data.results);
```
Comment (LoicBrassart) [inline src/components/FiltersTypes.jsx]: I'd rather sanitize values here

File: src/components/Pokemon.jsx [hunk 5]
```diff
@@ -27,56 +40,132 @@ const Pokemon = ({ url }) => {
...
+    if (heightFilters === 'small') {
+      return infos.height < 10;
+    }
+    if (heightFilters === 'medium') {
+      return infos.height >= 10 && infos.height <= 20;
+    }
+    if (heightFilters === 'big') {
+      return infos.height > 20;
+    }
+    return true;
```
Comment (LoicBrassart) [inline src/components/Pokemon.jsx:101]: Maybe use `switch...case` ? :smirk:

File: src/components/Pokemon.jsx [hunk 6]
```diff
@@ -27,56 +40,132 @@ const Pokemon = ({ url }) => {
...
+    if (weightFilters === 'light') {
+      return infos.weight < 1000;
+    }
+    if (weightFilters === 'medium') {
+      return infos.weight >= 1000 && infos.weight <= 3000;
+    }
+    if (weightFilters === 'heavy') {
+      return infos.weight > 3000;
+    }
+    return true;
```
Comment (LoicBrassart) [inline src/components/Pokemon.jsx:114]: Here, too

File: src/components/Pokemon.jsx [hunk 7]
```diff
@@ -27,56 +40,132 @@ const Pokemon = ({ url }) => {
...
+            <div className="pokemon-infos">
+              <h2 className="pokemon-name">
+                {infos.name.charAt(0).toUpperCase() + infos.name.slice(1)}
+              </h2>
+              <p className="pokemon-id">
+                #
+                {infos.id.toLocaleString('en-US', {
+                  minimumIntegerDigits: 3,
+                  useGrouping: false,
+                })}
```
Comment (LoicBrassart) [inline src/components/Pokemon.jsx]: Did you really need this ? o.O

File: src/components/PokemonList.jsx [hunk 8]
```diff
@@ -1,33 +1,55 @@
...
+  weightFilters,
+}) => {
   const [pokemons, setPokemons] = useState([]);
 
-  /* Call the API to get the list of pokémons that we want to display on the
-    Homepage (here the 50 first Pokemon) */
   useEffect(() => {
     axios
-      .get('https://pokeapi.co/api/v2/pokemon?limit=50&offset=0')
+      .get('https://pokeapi.co/api/v2/pokemon?limit=100&offset=650')
```
Comment (LoicBrassart) [inline src/components/PokemonList.jsx]: Why do you display these specific 100 Pokemon, then ?

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Some Evils, but globally good ;-)

File: src/components/Filters.jsx (hunk 1 above)
Comment (bdtkevin) [reply src/components/Filters.jsx]: 👍

File: src/components/Filters.jsx (hunk 2 above)
Comment (bdtkevin) [reply src/components/Filters.jsx]: 👍

File: src/components/FiltersAbilities.jsx (hunk 3 above)
Comment (bdtkevin) [reply src/components/FiltersAbilities.jsx]: 👍

File: src/components/FiltersTypes.jsx (hunk 4 above)
Comment (bdtkevin) [reply src/components/FiltersTypes.jsx]: 👍

File: src/components/Pokemon.jsx (hunk 5 above)
Comment (bdtkevin) [reply src/components/Pokemon.jsx:101]: Impossible with the custom repository 😭

File: src/components/Pokemon.jsx (hunk 6 above)
Comment (bdtkevin) [reply src/components/Pokemon.jsx:114]: Impossible with the custom repository 😭

File: src/components/PokemonList.jsx (hunk 8 above)
Comment (bdtkevin) [reply src/components/PokemonList.jsx]: It's just a random pick for the test, I have change for the first 100 now 🙂

File: src/components/Pokemon.jsx (hunk 7 above)
Comment (bdtkevin) [reply src/components/Pokemon.jsx]: Yes for a better display of id on the Pokemon component, I have move this in the .then of axios call

File: src/components/Pokemon.jsx [hunk 9]
```diff
@@ -132,11 +137,11 @@ const Pokemon = ({
...
-                #
-                {infos.id.toLocaleString('en-US', {
-                  minimumIntegerDigits: 3,
-                  useGrouping: false,
-                })}
+                {`#${infos.id}`}
+                {/* // {infos.id.toLocaleString('en-US', {
+                //   minimumIntegerDigits: 3,
+                //   useGrouping: false,
+                // })} */}
```
Comment (LoicBrassart) [inline src/components/Pokemon.jsx]: ... Dead code =D

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Still nope !

**Review by RobinDela — DISMISSED** [review]
All good.

File: src/components/Pokemon.jsx (hunk 9 above)
Comment (bdtkevin) [reply src/components/Pokemon.jsx]: NOOOOO

**Review by RobinDela — APPROVED** [review]
Approved

**Review by LoicBrassart — APPROVED** [review]
No more Evil

**Pull request merged**